aptos-move/framework/src/natives/transaction_context.rs

```diff
@@ -34,6 +40,43 @@ impl NativeTransactionContext {
     }
 }
 
+/***************************************************************************************************
+ * native fun create_uuid
+ *
+ *   gas cost: base_cost
+ *
+ **************************************************************************************************/
+#[derive(Clone, Debug)]
+pub struct CreateUUIDGasParameters {
+    pub base: InternalGas,
+}
+
+fn native_create_uuid(
+    gas_params: &CreateUUIDGasParameters,
+    context: &mut SafeNativeContext,
+    mut _ty_args: Vec<Type>,
+    _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    context.charge(gas_params.base)?;
```

Comment:
You could, sure! Just pass those SHA256 gas parameter along as an extra parameter when the native is created. We have an example of this in the `algebra.move` module.

Reply:
This would be a cleaner solution and would also address Alin's comment on the redundant gas schedule entry

Reply:
I talked to @vgao1996 today about both the options. He said "Both would work. Although I'd personally prefer the new gas parameter approach to decouple create_uuid from sha256."

Reply:
Yeah, my concern was is `create_uuid` and `sha256` are conceptually separate things, and reusing the sha256 gas parameters for create_uuid seems like a leaky abstraction to me, as implies the `create_uuid` is implemented using sha256 under the hood. Not totally sure if this is possible at all, but what if we change the hashing algorithm in the future?

Reply:
Can we separate the gas charging in sha256 into a common function that this can call? This is effectively doing a sha256 of the tx hash. If we ever need to change the cost of sha256 in the future, we'd need to remember there are two places to change if we duplicate the cost formula here

Reply:
@vgao1996, I am not sure I follow. Since the `create_uuid` calls an operation that does SHA2-256 it seems perfectly reasonable to me (and more maintainable) to pass the SHA2-256 gas parameters in to that native, as opposed to duplicating their costs.

If we change the hashing algorithm in the future, we can update the native to take those new, potentially-different gas parameters as input, no?

Reply:
Well, I defer to you if you think *reusing* gas parameters might be a good idea. My first instinct is that duplicating gas parameters is likely to lead to forgetting to update them in all the places. Which is why I think we should re-use them....

Reply:
Is the point here that we should be charging here cost of SHA2-256 + extra ? is that possible to specify, and have only extra be configurable?

Reply:
@alinush With the updated code, then `UUID = Hash ( Hash(TransactionDerivedUUID) || 0xFB )`. So, there are two hashes right? So, we should be charging more for UUID generation?

Reply:
Correct, but like @igor-aptos is saying below, the `CryptoHasher` domain separation is not actually necessary. This is true AFAICT. So we can just do `UUID = Hash ( Bytes(TransactionDerivedUUID) || 0xFB )`